src/Workspaces/Core/Portable/Utilities/NullableHelpers/TypeSymbolWithNullableAnnotation.cs

```diff
@@ -25,11 +25,21 @@ protected TypeSymbolWithNullableAnnotation(ITypeSymbol wrappedSymbol, NullableAn
             }
 
             public bool Equals(ISymbol other)
+            {
+                var t2 = other as ITypeSymbol;
+                if (t2 is null)
+                {
+                    return false;
+                }
```

Comment:
Minor point: This looks unnecessary since the overload below already handles `null` and takes an `ISymbol`.

Reply:
We still need it because that implements ITypeSymbol directly, and thus inherits the IEquatable interface. I'll make it explicit too

---
In reply to: [306024741](https://github.com/dotnet/roslyn/pull/37247#discussion_r306024741) [](ancestors = 306024741)